### ISSUE TITLE
Softmax layer unexpected behaviour for axis=0

### ISSUE THREAD

**Issue by rschumi0** [tensorflow/tensorflow#48647]
### Description:
The [Softmax layer](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Softmax) returns all ones when axis=0 is used. Perhaps this behaviour is not wrong, but it is really confusing since it is not what the user would expect from this layer. 
It would make sense to better explain in the documentation how the axis are numbered, especially since both negative and positive numbers can be used. Moreover, it might also make sense to add an error or warning message when axis=0 is used.

### Usage example
```
import tensorflow as tf
from tensorflow import keras
model = keras.Sequential([
keras.layers.Softmax(axis=0, input_shape=(3, 3))])
x = tf.constant([[[0.8055, 0.0083, 0.4057], [0.1249, 0.9762, 0.5402], [0.0637, 0.1539, 0.0282]]])
print (model.predict(x,steps=1))
```
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.4.0-rc4-71-g582c8d236cb 2.4.0
- Python version: 3.8.5

**Comment by szutenberg** [comment]
```
x = tf.Tensor(
[[[0.8055 0.0083 0.4057]
  [0.1249 0.9762 0.5402]
  [0.0637 0.1539 0.0282]]], shape=(1, 3, 3), dtype=float32)
```

You do normalization along axis = 0 (first dimension which is 1) so you get `exp(x)/exp(x) = 1`.

By default, softmax does normalization along the last dimension (axis = -1). 

I recommend [Introduction to Tensors - About shapes](https://www.tensorflow.org/guide/tensor?hl=en#about_shapes). Please send PR if it can be improved.

**Comment by jvishnuvardhan** [comment]
@szutenberg Thanks for the answer. 

@rschumi0 Please close the issue if this was resolved for you. Thanks!

**Comment by rschumi0** [comment]
@szutenberg  Thank you for your answer. It's true, one is the correct result when the softmax function is applied for axis=0. I still think that the user wouldn't use this layer to turn all inputs to one since this is usually not intended . I think this might happen by accident when users are not aware about the axis notations. Hence, I would suggest a warning in order to avoid unnecessary debugging efforts.

**Comment by jvishnuvardhan** [comment]
@rschumi0 Do you want to update any TF docs with a note to users? Please feel free to create any PR to update the docs. Thanks!
